### PR TITLE
added functionality for adding user and auth classes in API

### PR DESCRIPTION
**Pull request by sammy-perry-kainos**

### CODE REVIEW

File: src/main/java/org/kainos/ea/api/AuthService.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            if (id == -1) {
+                throw new FailedToCreateUserException();
+            }
+
+            return id;
+
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+
+            throw new RuntimeException(e);
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/api/AuthService.java]: We shouldn't throw a runtime exception here

File: src/main/java/org/kainos/ea/resources/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @POST
+    @Path("/login")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response login(Login login) {
+        try {
+            return Response.ok(authService.login(login)).build();
+        } catch (FailedToLoginException e) {
+            System.err.println(e.getMessage());
+
+            return Response.status(Response.Status.BAD_REQUEST).entity(e.getMessage()).build();
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/resources/AuthController.java]: Is 400 the correct status code for a failed login?

File: src/main/java/org/kainos/ea/api/AuthService.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        }
+
+        return false;
+    }
+
+    public boolean isUser(String token) throws TokenExpiredException, FailedToVerifyTokenException {
+        try {
+            int roleID = authDao.getRoleIDFromToken(token);
+
+            if (roleID == 2) {
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/api/AuthService.java]: Is there something we could use that is easier to understand when reading the code apart from a number

Comment (sammy-perry-kainos) [reply src/main/java/org/kainos/ea/api/AuthService.java]: updated

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**